### ISSUE TITLE
tensor_diag_part does not vectorize?

### ISSUE THREAD

**Issue by ziofil** [tensorflow/tensorflow#34734]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): MacOS 10.15.1
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.0.0
- Python version: 3.7

**Describe the current behavior**
This code throws an error, despite `tf.linalg.tensor_diag_part` works on each of the two matrices:

```python
    k = tf.convert_to_tensor(np.arange(8).reshape((2,2,2)))
    tf.vectorized_map(tf.linalg.tensor_diag_part, k)
```

> UnrecognizedFlagError: Unknown command line flag 'f'

It should return the diagonal of each of the two submatrices.

**Comment by ravikyram** [comment]
I have tried on colab with TF version 2.0, 2.1.0-dev20191201 and was able to reproduce the issue.Please, find the gist [here](https://colab.sandbox.google.com/gist/ravikyram/fecdecc2fcfd23ef317022ead88bef0d/untitled433.ipynb). Thanks!

**Comment by ziofil** [comment]
Hello, any progress? I have found this is not the only case, see https://github.com/tensorflow/tensorflow/issues/35579

**Comment by ymodak** [comment]
This is fixed with tf-nightly version '2.2.0-dev20200218' .Thanks!
